### PR TITLE
[DOC-9552] Document limitations on folder names

### PR DESCRIPTION
**Pull request by mdlinville**
This is the PR to document the limitations that are deployed in Production today

### CODE REVIEW

**Review by Amruta-Ranade — COMMENTED** [review]
This PR seems like a subset of https://github.com/cockroachdb/docs/pull/18239/
What am I missing?

**Comment by mdlinville** [comment]
> This PR seems like a subset of #18239 What am I missing?

This one needs to go in first, to document the undocumented rules about folder names today. The apostrophe support will be in CC next week, and #18239 will go in then.

**Review by Amruta-Ranade — APPROVED** [review]

File: src/current/cockroachcloud/folders.md [hunk 1]
```diff
@@ -27,6 +27,11 @@ To move a folder or a cluster, you update its `parent_id`. If you move a folder
 - A violation of the [folder structure](#folder-structure) limitations.
 - An attempt to move a folder into itself or into one of its descendant folders.
 
+### Folder naming
+
+- Folder names must begin and end with a letter or number. The only allowed special characters are spaces, hyphens, and underscores.
+- Folder names cannot exceed 40 characters.
+
```
Comment (xhesikam) [inline src/current/cockroachcloud/folders.md:34]: Looks good! Can you please include that folder names must be at least 3 characters.

File: src/current/cockroachcloud/folders.md [hunk 2]
```diff
@@ -242,6 +247,7 @@ To delete a folder:
 
 ## Limitations
 
+- Folder names must begin and end with a letter or number. The only allowed special characters are spaces, hyphens, and underscores. In addition, folder names cannot exceed 40 characters.
```
Comment (xhesikam) [inline src/current/cockroachcloud/folders.md:250]: Same thing here. Folder names should be at least 3 characters.

**Review by xhesikam — CHANGES_REQUESTED** [review]

File: src/current/cockroachcloud/folders.md [hunk 3]
```diff
@@ -27,6 +27,11 @@ To move a folder or a cluster, you update its `parent_id`. If you move a folder
 - A violation of the [folder structure](#folder-structure) limitations.
 - An attempt to move a folder into itself or into one of its descendant folders.
 
+### Folder naming
+
+- Folder names must begin and end with a letter or number. The only allowed special characters are spaces, hyphens, and underscores.
+- Folder names  must be at least 3 characters long and cannot exceed 40 characters.
```
Comment (mdlinville) [inline src/current/cockroachcloud/folders.md:33]:
```suggestion
- Folder names must be at least 3 characters long and cannot exceed 40 characters.
```

**Review by xhesikam — APPROVED** [review]

**Pull request merged**